Fixed quest 3058 cannot be completed properly

sky1683589933 opened the pull request:
Fixed issue with dialog,Add animations for item usage and reject quest options.

sky1683589933 committed:
Fixed 3058 quest cannot be completed properlyges

Fixed issue with dialog,Add animations for item usage and reject quest options.

sky1683589933 committed:
Merge branch 'beyond-aion:4.8' into Pull-Request_test

sky1683589933 committed:
Merge branch 'beyond-aion:4.8' into Pull-Request_test

sky1683589933 committed:
Fixed quest 3058 cannot be completed properly

Fixed issue with dialog,Add animations for item usage and reject quest options.

neon-dev commented on game-server/data/handlers/quest/theobomos/_3058StoneofMabolo.java:
```diff
@@ -68,10 +72,7 @@ else if (env.getDialogActionId() == SELECT_QUEST_REWARD)
 					return false;
 				case SETPRO1:
 					if (var == 0) {
```
You don't need to check the `var` here anymore, since you're using `defaultCloseDialog(env, 0, 1)` which checks the quest var internally.
This simplifies the whole `SETPRO1` case to a single return statement.

sky1683589933 replied:
Okay, I understand. I will pay attention to other modifications next time because this local branch has already been deleted.

neon-dev replied:
You can checkout the branch locally again, as long as it is not deleted on the remote origin (GitHub in this case).
You can also edit files directly in the browser:
![grafik](https://github.com/user-attachments/assets/af27919f-976f-4236-8fc0-da987e19dca5)

neon-dev commented on game-server/data/handlers/quest/theobomos/_3058StoneofMabolo.java:
```diff
@@ -83,15 +84,30 @@ else if (env.getDialogActionId() == SELECT_QUEST_REWARD)
 					return false;
 				case SETPRO2:
 					if (var == 1) {
```
You don't need to check the `var` here anymore, since you're using `defaultCloseDialog(env, 1, 2)` which checks the quest var internally.
This simplifies the whole `SETPRO2` case to a single return statement.

sky1683589933 committed:
Fixed quest 3058 cannot be completed properly

Fixed issue with dialog,Add animations for item usage and reject quest options.

sky1683589933 committed:
Fixed quest 3058 cannot be completed properly

Fixed issue with dialog,Add animations for item usage and reject quest options.

neon-dev committed:
Adjustments based on retail info

neon-dev reviewed (approved):
I have made some changes based on this video I found: https://www.youtube.com/watch?v=x3QQfsMq7LM&t=12s
The quest start item now uses the correct animation packet and the last quest step now switches to the reward state while talking to Zirius.
Please test my changes and let me know if you think this pull request can be merged.

sky1683589933 commented:
> I have made some changes based on this video I found: https://www.youtube.com/watch?v=x3QQfsMq7LM&t=12s The quest start item now uses the correct animation packet and the last quest step now switches to the reward state while talking to Zirius. Please test my changes and let me know if you think this pull request can be merged.

I have tested that this is a very good change, everything is perfect!
Adding items to the kernel and automatically matching animations of using items through the client is amazing!
